grpc/src/main/java/io/stargate/grpc/service/SynchronizedStreamObserver.java

```diff
@@ -0,0 +1,49 @@
+package io.stargate.grpc.service;
+
+import io.grpc.stub.StreamObserver;
+
+/**
+ * It wraps the underlying {@link StreamObserver} and delegates all the method to it. Additionally,
+ * it adds a synchronization on {@code this} because, as stated in the <a
+ * href="https://grpc.github.io/grpc-java/javadoc/io/grpc/stub/StreamObserver.html">StreamObserver
+ * docs</a>:
+ *
+ * <pre>
+ * Since individual StreamObservers are not thread-safe,
+ * if multiple threads will bewriting to a StreamObserver concurrently,
+ * the application must synchronize calls.
+ * </pre>
+ *
+ * all calls need to be synchronized.
+ *
+ * @param <V>
+ */
+public class SynchronizedStreamObserver<V> implements StreamObserver<V> {
+
+  private final StreamObserver<V> streamObserver;
+
+  public SynchronizedStreamObserver(StreamObserver<V> streamObserver) {
+    this.streamObserver = streamObserver;
+  }
+
+  @Override
+  public void onNext(V value) {
+    synchronized (this) {
```

Comment:
I would say that `synchronized(streamObserver)` is better option here.. I know this will work as well, but with my suggestion you are syncing on the `streamObserver` and you would be respecting anybody from outside this class doing the same.. I know this should not happen, but seems like a more valid design for me? wdyt?

if you accept this, of course change it everywhere and update that class level java doc